streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -295,52 +369,79 @@ public void testMaybePunctuate() throws Exception {
                 new ConsumerRecord<>(partition2.topic(), partition2.partition(), 45, 0L, TimestampType.CREATE_TIME, 0L, 0, 0, recordKey, recordValue)
         ));
 
-        assertTrue(task.maybePunctuate());
+        assertTrue(task.maybePunctuateStreamTime());
 
         assertTrue(task.process());
         assertEquals(5, task.numBuffered());
         assertEquals(1, source1.numReceived);
         assertEquals(0, source2.numReceived);
 
-        assertFalse(task.maybePunctuate());
+        assertFalse(task.maybePunctuateStreamTime());
 
         assertTrue(task.process());
         assertEquals(4, task.numBuffered());
         assertEquals(1, source1.numReceived);
         assertEquals(1, source2.numReceived);
 
-        assertTrue(task.maybePunctuate());
+        processorStreamTime.supplier.scheduleCancellable.cancel();
+
+        assertFalse(task.maybePunctuateStreamTime());
 
         assertTrue(task.process());
         assertEquals(3, task.numBuffered());
         assertEquals(2, source1.numReceived);
         assertEquals(1, source2.numReceived);
 
-        assertFalse(task.maybePunctuate());
+        assertFalse(task.maybePunctuateStreamTime());
 
         assertTrue(task.process());
         assertEquals(2, task.numBuffered());
         assertEquals(2, source1.numReceived);
         assertEquals(2, source2.numReceived);
 
-        assertTrue(task.maybePunctuate());
+        assertFalse(task.maybePunctuateStreamTime());
 
         assertTrue(task.process());
         assertEquals(1, task.numBuffered());
         assertEquals(3, source1.numReceived);
         assertEquals(2, source2.numReceived);
 
-        assertFalse(task.maybePunctuate());
+        assertFalse(task.maybePunctuateStreamTime());
 
         assertTrue(task.process());
         assertEquals(0, task.numBuffered());
         assertEquals(3, source1.numReceived);
         assertEquals(3, source2.numReceived);
 
         assertFalse(task.process());
-        assertFalse(task.maybePunctuate());
+        assertFalse(task.maybePunctuateStreamTime());
 
-        processor.supplier.checkAndClearPunctuateResult(20L, 30L, 40L);
+        processorStreamTime.supplier.checkAndClearPunctuateResult(PunctuationType.STREAM_TIME, 20L);
+    }
+
+    @Test
+    public void testMaybePunctuateSystemTime() throws Exception {
```

Comment:
Also we should probably have another test: `shouldNotPunctuateSystemTimeWhenIntervalNotElapsed`